packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -327,12 +327,50 @@ class RenderSliverMainAxisGroup extends RenderSliver with ContainerRenderObjectM
 
   @override
   void paint(PaintingContext context, Offset offset) {
-    RenderSliver? child = lastChild;
+    if (firstChild == null) {
+      return;
+    }
+    // offset is to the top-left corner, regardless of our axis direction.
+    // originOffset gives us the delta from the real origin to the origin in the axis direction.
+    final Offset mainAxisUnit, crossAxisUnit, originOffset;
+    final bool addExtent;
+    switch (applyGrowthDirectionToAxisDirection(constraints.axisDirection, constraints.growthDirection)) {
+      case AxisDirection.up:
+        mainAxisUnit = const Offset(0.0, -1.0);
+        crossAxisUnit = const Offset(1.0, 0.0);
+        originOffset = offset + Offset(0.0, geometry!.paintExtent);
+        addExtent = true;
+      case AxisDirection.right:
+        mainAxisUnit = const Offset(1.0, 0.0);
+        crossAxisUnit = const Offset(0.0, 1.0);
+        originOffset = offset;
+        addExtent = false;
+      case AxisDirection.down:
+        mainAxisUnit = const Offset(0.0, 1.0);
+        crossAxisUnit = const Offset(1.0, 0.0);
+        originOffset = offset;
+        addExtent = false;
+      case AxisDirection.left:
+        mainAxisUnit = const Offset(-1.0, 0.0);
+        crossAxisUnit = const Offset(0.0, 1.0);
+        originOffset = offset + Offset(geometry!.paintExtent, 0.0);
+        addExtent = true;
+    }
 
+    RenderSliver? child = lastChild;
```

Comment:
@Piinks Thank you for working on this!

Note, I'm not well versed with slivers so I might be wrong here, but wouldn't this paint children in reverse order?
Wouldn't a layout like this be rendered incorrectly then? i.e. the blue box would be rendered beneath the brown box due to a paint transform being applied.
```dart
SliverMainAxisGroup(slivers: [
  const SliverToBoxAdapter(
    child: ColoredBox(
      color: Colors.brown,
      child: SizedBox(height: 100),
    ),
  ),
  SliverToBoxAdapter(
    child: Transform.translate(
      offset: const Offset(0, -50),
      child: const ColoredBox(
        color: Colors.blue,
        child: SizedBox(height: 100),
      ),
    ),
  ),
]);
```

However, it seems slivers in general are painted in reverse.

I tested the following code on stable and I'd assume all to render the same but they don't.
`PageThroughList` renders as expected, while `MainAxisGroupPage` and `PageThroughSlivers` render the blue box below the brown one.
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: MainAxisGroupPage(),
    );
  }
}

class MainAxisGroupPage extends StatelessWidget {
  const MainAxisGroupPage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: CustomScrollView(slivers: [
        SliverMainAxisGroup(slivers: [
          const SliverToBoxAdapter(
            child: ColoredBox(
              color: Colors.brown,
              child: SizedBox(height: 100),
            ),
          ),
          SliverToBoxAdapter(
            child: Transform.translate(
              offset: const Offset(0, -50),
              child: const ColoredBox(
                color: Colors.blue,
                child: SizedBox(height: 100),
              ),
            ),
          ),
        ]),
      ]),
    );
  }
}

class PageThroughSlivers extends StatelessWidget {
  const PageThroughSlivers({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: CustomScrollView(slivers: [
        const SliverToBoxAdapter(
          child: ColoredBox(
            color: Colors.brown,
            child: SizedBox(height: 100),
          ),
        ),
        SliverToBoxAdapter(
          child: Transform.translate(
            offset: const Offset(0, -50),
            child: const ColoredBox(
              color: Colors.blue,
              child: SizedBox(height: 100),
            ),
          ),
        ),
      ]),
    );
  }
}

class PageThroughList extends StatelessWidget {
  const PageThroughList({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: CustomScrollView(slivers: [
        SliverList.list(children: [
          const ColoredBox(
            color: Colors.brown,
            child: SizedBox(height: 100),
          ),
          Transform.translate(
            offset: const Offset(0, -50),
            child: const ColoredBox(
              color: Colors.blue,
              child: SizedBox(height: 100),
            ),
          ),
        ]),
      ]),
    );
  }
}
```
Should I file another issue or is this working as expected?